### PR TITLE
Switch to Glide

### PR DESCRIPTION
**Pull request by hfatih**
Migrated to glide for better performance and cache management, and built in gif support to be able to use gifs on any ImageView. Also made ImageDialog full screen, with gif support and panning/zooming. 

I used application class for centralized image loading, been using this approach with no issues, if there is any caveat please let me know.

### CODE REVIEW

File: WhatAndroid/src/main/java/what/whatandroid/login/LoggedInActivity.java [hunk 1]
```diff
@@ -62,34 +55,7 @@
 	private Login loginTask;
 	private LogoutTask logoutTask;
 
-	/**
-	 * Initialize MySoup so that we can start making API requests
-	 */
-	public static void initSoup(){
```
Comment (Twinklebear) [inline WhatAndroid/src/main/java/what/whatandroid/login/LoggedInActivity.java]: Why was this removed? The way the WhatAPI works is that we need to set some static information like the site name, if we're using SSL and our user agent. Without this we won't be able to make API requests to the site. Related to this the site string shouldn't have been removed either.

I haven't had time to fully go through the PR and try testing it but this stuck out.

Comment (Twinklebear) [inline WhatAndroid/src/main/java/what/whatandroid/login/LoggedInActivity.java]: Ah I see this moved up to WhatApplication

File: WhatAndroid/src/main/java/what/whatandroid/WhatApplication.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import what.whatandroid.imgloader.ImageLoadFailTracker;
+
+/**
+ * Created by fatih on 11.9.2015.
+ */
+public class WhatApplication extends Application {
+
+    //TODO: Developers put your local Gazelle install IP here instead of testing on the live site
+    //I recommend setting up with Vagrant: https://github.com/dr4g0nnn/VagrantGazelle
+    public static final String DEFAULT_SITE = "https://what.cd/";
```
Comment (Twinklebear) [inline WhatAndroid/src/main/java/what/whatandroid/WhatApplication.java]: You shouldn't put the site URL here, put the same local vagrant one.

**Comment by Twinklebear** [comment]
Still working through the Pull Request, but why did you split parts of LoggedInActivity into WhatApplication? It seems like it's just a container for some static methods so why would it inherit from Application if we won't actually instantiate it (do we? I'm still reading).

**Comment by hfatih** [comment]
A normal class with some static methods/variables is not guaranteed to live through application lifecycle. It can be killed, variables may get removed etc. Application class is guaranteed to live. By adding it into the manifest as the name of the application, it is guaranteed that WhatApplication class initializes first, before any acticity or class, and it is killed last. 

For these properties, i always thought doing app-wide things and initializations in Application class is beneficial, or convenient at least. And had no problems in the past with it.

**Comment by Twinklebear** [comment]
Hmm, I see. I also noticed you took out checking for null or empty image URLs which we do sometimes get on the site. Will Glide just throw an exception which is picked up by the exception handler given to it?

File: WhatAndroid/src/main/java/what/whatandroid/views/TouchImageView.java [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+		super(context);
+		sharedConstructing(context);
+	}
+
+	public TouchImageView(Context context, AttributeSet attrs) {
+		super(context, attrs);
+		sharedConstructing(context);
+	}
+
+	private void sharedConstructing(Context context) {
```
Comment (Twinklebear) [inline WhatAndroid/src/main/java/what/whatandroid/views/TouchImageView.java]: For readability I'd prefer it if there wasn't a newline after every statement here but that things were grouped logically instead.

Comment (hfatih) [inline WhatAndroid/src/main/java/what/whatandroid/views/TouchImageView.java]: Sure, i will refactor it.

**Comment by hfatih** [comment]
Yes, our exception handler catches it and shows the appropriate error image/placeholder.

**Comment by stuxo** [comment]
Should this be PR be targeting develop rather than master?

**Comment by Twinklebear** [comment]
@stuxo You're right, good catch. @hfatih Could you change this pull to be targeting the develop branch?

**Comment by hfatih** [comment]
Oh, my bad. I didn't even use development branch to begin with. I will reapply these changes to development branch and open a new pull request if we are okay with the changes.

**Comment by Twinklebear** [comment]
@hfatih do you think you may have time to finish up this PR? We're hoping to get a release out soon with the material design update and some bug fixes and this would be awesome PR to have in as well. If you don't think you'd have time I can finish making the final tweaks and testing to merge this in if it's in an ok state.

**Comment by hfatih** [comment]
Sure, i already applied little changes mentioned above and tested everywhere image loading used.  Gifs work everywhere too. I will make another pull request in dev branch tonight.
